doc/man3/SSL_CTX_set_timeout.pod

```diff
@@ -2,21 +2,27 @@
 
 =head1 NAME
 
-SSL_CTX_set_timeout, SSL_CTX_get_timeout - manipulate timeout values for session caching
+SSL_CTX_set_timeout, SSL_CTX_get_timeout,
+SSL_CTX_set_timeout_ex,
+SSL_CTX_get_timeout_ex - manipulate timeout values for session caching
 
 =head1 SYNOPSIS
 
  #include <openssl/ssl.h>
 
  long SSL_CTX_set_timeout(SSL_CTX *ctx, long t);
  long SSL_CTX_get_timeout(SSL_CTX *ctx);
+ time_t SSL_CTX_set_timeout_ex(SSL_CTX *ctx, time_t t);
```

Comment:
The timeout is a duration isn't it.  As such, I don't see a need to accept a `time_t`.  `long` is more than sufficient.

Reply:
Duration and let call it "calandar time" are related but distinct concepts. Duration is not fixed in time, whereas a calendar time is. However, calendar time is nothing but a duration from a fixed point in time. So, they really could be the same type. Both are effectively bound by the same constraints, one of which is the size/type of storage. The OSSL_TIME code itself allows for subtraction (which results in a duration) and addition (one of which must be a duration). `sleep(3)` takes a duration in seconds, as an `unsigned int`. But I swear there was a type specific for durations, but I can't find it.

Reply:
There are more than a few variations documented:

| Call | Type |
| --- | ---: |
| `alarm(2)` | `unsigned int` |
| `clock(2)` | `clock_t` |
| `clock_gettime(2)` | `struct timespec` |
| `epoll_wait(2)` | `int` |
| getitimer(2)` | `struct timeval` |
| `poll(2)` | `int` |
| `sleep(3)` | `unsigned int` |
| `usleep(3)` | `useconds_t` |

I can't locate a specific duration type either.